src/main/java/me/ryanhamshire/GriefPrevention/BlockEventHandler.java

```diff
@@ -232,6 +232,24 @@ public void onBlockPlace(BlockPlaceEvent placeEvent)
 		String noBuildReason = GriefPrevention.instance.allowBuild(player, block.getLocation(), block.getType());
 		if(noBuildReason != null)
 		{
+			// Prevent players without container trust from putting books into lecterns
+			Material mainHand = player.getInventory().getItemInMainHand().getType();
+			Material offHand = player.getInventory().getItemInOffHand().getType();
+			PlayerData playerData = this.dataStore.getPlayerData(player.getUniqueId());
+			Claim claim = this.dataStore.getClaimAt(block.getLocation(), true, playerData.lastClaim);
+			if (block.getType() == Material.LECTERN && mainHand != Material.LECTERN && offHand != Material.LECTERN)
```

Comment:
Shouldn't the hand material be checked against `BlockPlaceEvent#getHand`? This check could fail when holding a lectern in one hand and a book in the other. Weird edge case, but technically could block valid book insertion.
Could check that `BlockPlaceEvent#getReplacedState`'s type is also a lectern instead of checking hands.

Reply:
Good call, Ill do some testing and make the changes.

Reply:
Well, that was a surprisingly easy fix.